src/main/java/seedu/task/logic/commands/EditTaskCommand.java

```diff
@@ -98,19 +98,26 @@ private Task editTask(ReadOnlyTask targetTask) {
         if (!isNameToBeEdit) {
             newName = targetTask.getTask();
         }
-        if (!isDescriptionToBeEdit) {
-            newDescription = targetTask.getDescription();
+        if (!isDescriptionToBeEdit && targetTask.getDescription().isPresent()) {
```

Comment:
Yea, this would make the task constructor more elegant. So instead of parsing the Name, Description, etc, you just pass strings in and create the Name, Description, etc classes in the constructor.